File: app/src/main/java/com/olehmesh/randomusers_task/common/UsersContract.java
package com.olehmesh.randomusers_task.common;

import com.olehmesh.randomusers_task.models.Result;

import java.util.List;

public class UsersContract {

  public interface View {

        void init();

        void showProgress();

        void hideProgress();

        void showError(String message);

        void loadDataInList(List<Result> users);

    }

  public interface Presenter {

        void onRefreshButtonClick();

        void start();

        void loadUsers();
    }

}

File: app/src/main/java/com/olehmesh/randomusers_task/common/UsersContract.kt
package com.olehmesh.randomusers_task.common

import com.olehmesh.randomusers_task.models.Result

class UsersContract {

    interface View {

        fun init()

        fun showProgress()

        fun hideProgress()

        fun showError(message: String)

        fun loadDataInList(users: List<Result>)

    }

    interface Presenter {

        fun onRefreshButtonClick()

        fun start()

        fun loadUsers()
    }

}
